unity-game/Assets/Scripts/PlayerController.cs

```diff
@@ -52,6 +52,8 @@ public class PlayerController : MonoBehaviour
     private bool interactedWithLadder;
     private bool usingDoor = false;
     private bool disableMovement;
+    private bool disableFixedUpdateMovement;
+    private bool disableUpdateMovement;
```

Comment:
disableMovement is used when insideSolidObject && ladderZone. That's what enables ladders to not glitch out when the player is inside of a solid object.

The other two are set up as precautionary bools. In case we want to limit the player's movement ability in specific ways.